src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -59,30 +60,56 @@ void check_sshfs_exists(mp::SSHSession& session)
     run_cmd(session, "which sshfs", error_handler);
 }
 
-void make_target_dir(mp::SSHSession& session, const std::string& target)
+// Split a path into existing and to-be-created parts. The input is assumed to
+// be an absolute path, normalized with mp::utils::normalize_absolute_path().
+std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
```

Comment:
Isn't this part of the other PR? In such occasions, where you need the changes of a PR that still hasn't gone in, it is better to stack the PRs. So you'd branch this PR out of where the other PR left off.

Reply:
Yes, that's part of the other PR. There can be a problem when merging we'd need to solve. That can be done. I can try to stack the PR's, would be a good idea (altough I'll need to force-push at some point).

Reply:
Right, it is generally a good idea to avoid commit overlap. So stacking branches/PRs decouples things and is usually better (even though GH is not very good dealing with those). 

On force-pushing, I know it is generally not advised, but it is pretty much part of our usual workflow for PR branches (like when rebasing on the tip of master). It is fine as long as no one depends on the changing refs, so we usually avoid depending on each other's branches and we're ok if we don't force-push master.

Reply:
Well, we can manually merge on this specific case, there will be at most one conflict and the file on this commit is what will prevail. Next time I'll pay attention to this issue.

About force pushing, I'm easy. Since I am the only one pushing to this branch, there will be no issues.

Reply:
Solved, now this branch is on top of the other PR.